banwebScrape.py

```diff
@@ -32,13 +74,44 @@ def initialize_logging_manager(log_file: str = None, log_level: str = INFO):
 
 
 def main():
+    # Get CLI args
+    args = form_cli_args()
     # Initialize logger
-    initialize_logging_manager()
+    if args.debug:
+        args.log_level = DEBUG
+    initialize_logging_manager(args.log_file, args.log_level)
     log = Logger("main")
     log.debug("Starting banwebScrape.py")
     files = File_Utils()
-    files.write_to_csv(Data_Gathering().get_course_object_list(),
-                       "./writer.csv")
+
+    # If output file's extension matches output_type extension
+    # clip it to prevent duplicate extension names.
+    if args.output.split('.')[-1] != str(args.output_type):
+        out_file = "{filename}.{extension}".format(filename=args.output,
+                                                   extension=args.output_type)
+    else:
+        out_file = args.output
+    log.info("Preparing to write data to {}".format(out_file))
+    if args.in_file:
+        # no scrape, load from file
+        if path.exists(args.in_file):
+            if args.in_file_type == Output_Types.CSV:
+                class_list = files.read_from_csv_to_Class_Obj_List(
+                    args.in_file)
+            elif args.in_file_type == Output_Types.JSON:
+                class_list = files.read_from_json_to_Class_Obj_List(
+                    args.in_file)
+        else:
+            log.error("File {} not found.".format(args.in_file))
+            exit(2)
+    else:
+        class_list = Data_Gathering().get_course_object_list()
+        log.info("Data finished writing")
+
+    if args.output_type == Output_Types.CSV:
+        files.write_to_csv(class_list, out_file)
```

Comment:
Would it be worth it to add `--quiet` and `--stdout` options that piped the output directly to stdout? Could then modify `custom_db_sync` to accept stdin.

Reply:
I think the current goal is for the scraper and django command to eventually "merge". Or in general for the workflow to eventually become:
- scrape data
- collect data into Django model objects as the script iterates
- Bulk create

We'll have to have a chat with how we want to handle things like edits/removal of classes since those are edge cases we're currently not ready for.

Anyway, this is outside of the scope of the PR and will have to be settled in later issues.

Reply:
👌